architecture/configuration.md

```diff
@@ -0,0 +1,58 @@
+# Spike Config
+
+Most of the inner workings of `spike` will never be open to configuration, we
+allow the users a limited interface for changing some of the default behavior of
+`spike` for a smoother experience.
+
+This configuration will be handled by a configuration object, that will exist
+under a `"spike"` key in the users `package.json` or a `spike.config.js` file in
+the source directory.
+
+
+## Scope
+
+The configuration file will allow you to change the following options:
+
+- Changing the build target directory. Optional, default to `out/`.
+- Changing the source directory. Optional, default to `cwd`.
```

Comment:
IMHO, I don't think we should allow this.

Allowing this to change makes file resolution a lot harder to manage. Next.js doesn't allow setting this, you either:

- Follow the file conventions in the root dir
- Follow the file conventions in the src dir

Reply:
I dont think we should allow changing either of these.